doc/contributing.md

```diff
@@ -0,0 +1,165 @@
+
+## Testing
+
+### Run the tests
+
+The tests for micro-sam are run with [pytest](https://docs.pytest.org/en/7.4.x/)
+
+To run the tests:
+```
+pytest
+```
+
+### Writing your own tests
+
+If you have written new code, you will need to write tests to go with it.
+
+#### Unit tests
+
+Unit tests are the preferred style of tests for user contributions. Unit tests check small, isolated parts of the code for correctness. If your code is too complicated to write unit tests easily, you may need to consider breaking it up into smaller functions that are easier to test.
+
+#### Tests involving napari
+
+In cases where tests *must* use the napari viewer, [these tips might be helpful](https://napari.org/stable/plugins/test_deploy.html#tips-for-testing-napari-plugins) (in particular, the `make_napari_viewer_proxy` fixture).
+
+These kinds of tests should be used only in limited circumstances. Developers are [advised to prefer smaller unit tests, and avoid integration tests](https://napari.org/stable/plugins/test_deploy.html#prefer-smaller-unit-tests-when-possible) wherever possible.
+
+#### Code coverage
+
+Pytest uses the [pytest-cov](https://pytest-cov.readthedocs.io/en/latest/) plugin to automatically determine which lines of code are covered by tests.
+
+A short summary report is printed to the terminal output whenever you run pytest. The full results are also automatically written to a file named `coverage.xml`.
+
+The [Coverage Gutters VSCode extension](https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters) is useful for visualizing which parts of the code need better test coverage. PyCharm professional [has a similar feature](https://www.jetbrains.com/pycharm/guide/tips/spot-coverage-in-gutter/), and you may be able to find similar tools for your preferred editor.
+
+We also use [codecov.io](https://app.codecov.io/gh/computational-cell-analytics/micro-sam) to display the code coverage results from our Github Actions continuous integration.
+
+## Open a pull request
+
+Once you've made changes to the code and written some tests to go with it, you are ready to [open a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests). You can [mark your pull request as a draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests) if you are still working on it, and still get the benefit of discussing the best approach with maintainers.
+
+Remember that typically changes to micro-sam are made branching off from the development branch. So, you will need to open your pull request to merge back into the `dev` branch [like this](https://github.com/computational-cell-analytics/micro-sam/compare/dev...dev).
+
+## Optional: Build the documentation
```

Comment:
Turns out, I don't know how to build the documentation locally.

I found these two files: [.github/doc_env.yaml](https://github.com/computational-cell-analytics/micro-sam/blob/master/.github/doc_env.yaml) and [.github/workflows/build_docs.yaml](https://github.com/computational-cell-analytics/micro-sam/blob/master/.github/workflows/build_docs.yaml). 

I'm not very familiar with [pdoc](https://pdoc.dev/) thought, and since you say "We use a custom build script for pdoc itself" [here](https://github.com/computational-cell-analytics/micro-sam/blob/83997ff4a471cd2159fda4e26d1445f3be79eb08/.github/workflows/build_docs.yaml#L43) I think it's safer if you write this section.

Reply:
I think we should add `pdoc` to `requirements-dev.txt`. It's lightweight, so I don't anticipate any problems with installing it via pip.
Then you can use https://github.com/computational-cell-analytics/micro-sam/blob/master/build_doc.py to build the documentation.
This will directly start a local server with the doc.

Maybe you can add one sentence on this here.

Reply:
Done!